app/assets/javascripts/application.js.erb

```diff
@@ -1,15 +1,30 @@
-// This is a manifest file that'll be compiled into application.js, which will include all the files
-// listed below.
+//-- copyright
+// OpenProject is a project management system.
+// Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
 //
-// Any JavaScript/Coffee file within this directory, lib/assets/javascripts, vendor/assets/javascripts,
-// or vendor/assets/javascripts of plugins, if any, can be referenced here using a relative path.
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License version 3.
 //
-// It's not advisable to add code directly here, but if you do, it'll appear at the bottom of the
-// the compiled file.
+// OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+// Copyright (C) 2006-2013 Jean-Philippe Lang
+// Copyright (C) 2010-2013 the ChiliProject Team
 //
-// WARNING: THE FIRST BLANK LINE MARKS THE END OF WHAT'S TO BE PROCESSED, ANY BLANK LINE SHOULD
-// GO AFTER THE REQUIRES BELOW.
```

Comment:
I'd at least try to keep something like

````
// TODO: get rid of the code below, this file should ideally only consist of "requires"
// see https://help.ful.link.com/why-to-not-include-code-here
````

I only know by chance that we are following bad practices here and if anyone asked me, I couldn't even tell him why, except that it is good to split stuff among several files.
I think there was a real framework-ish reason.

I am pointing that out, because people on the code will change and they should at least have the chance to learn that stuff...